src/ucp/proto/proto_perf.h

```diff
@@ -55,43 +89,34 @@ void ucp_proto_perf_destroy(ucp_proto_perf_t *perf);
 
 
 /**
- * Create proto perf structure from proto caps.
- *
- * @param [in]  name        Name of the performance data structure.
- * @param [in]  proto_caps  Proto caps to create perf from.
- * @param [out] perf_p      Filled with the new performance data structure.
-*/
-ucs_status_t ucp_proto_perf_from_caps(const char *name,
-                                      const ucp_proto_caps_t *proto_caps,
-                                      ucp_proto_perf_t **perf_p);
+ * @return Whether the perf is empty.
+ */
+int ucp_proto_perf_is_empty(const ucp_proto_perf_t *perf);
 
 
 /**
  * Add linear functions to several performance factors at the range
  * [ @a start, @a end ]. The performance functions to add are provided in the
  * array @a funcs, each entry corresponding to a factor id defined in
- * @ref ucp_proto_perf_factor_id_t. The bitmap @a factors_bitmap specifies the
- * valid elements in that array; any elements not specified in the bitmap are
- * ignored.
+ * @ref ucp_proto_perf_factor_id_t.
  *
  * Initially, all ranges are uninitialized; repeated calls to this function
  * should be used to populate the @a perf data structure.
  *
  * @param [in] perf            Performance data structure to update.
  * @param [in] start           Add the performance function to this range start (inclusive).
  * @param [in] end             Add the performance function to this range end (inclusive).
- * @param [in] funcs           Array of performance functions to add.
- * @param [in] factors_bitmap  Bitmap of performance factors to add, using bit
- *                             indexes defined in @ref ucp_proto_perf_factor_id_t.
+ * @param [in] perf_factors    Array of performance functions to add.
```

Comment:
`child_perf_node` and `title` are missing

Reply:
Done